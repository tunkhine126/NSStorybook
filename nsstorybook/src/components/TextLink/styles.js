import { makeStyles } from '@material-ui/core/styles';
import colors from '../../global-styles';

export const textLinkStyles = makeStyles(() => ({
  btn: {
    padding: 0,
    fontSize: '12px',
    textAlign: 'inherit',
  },
}));

export const btnStyles = fontWeight =>
  makeStyles(() => ({
    label: {
      fontWeight: fontWeight ? '700' : 'normal',
    },
    textPrimary: {
      '&:hover': {
        backgroundColor: 'transparent',
        color: `${colors.LINKS.hover}`,
        transition: 'color 0.2s',
      },
    },
  }));
