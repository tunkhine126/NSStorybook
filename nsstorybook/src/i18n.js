/**
 * i18n.js
 *
 * This will setup the i18n language files and locale data for your app.
 *
 *   IMPORTANT: This file is used by the internal build
 *   script `extract-intl`, and must use CommonJS module syntax
 *   You CANNOT use import/export in this file.
 */
const addLocaleData = require('react-intl').addLocaleData; //eslint-disable-line
const enLocaleData = require('react-intl/locale-data/en');
const esLocaleData = require('react-intl/locale-data/es');
const ptLocaleData = require('react-intl/locale-data/pt');

// Custom Locale-Data for Haitian / Creole (not supported by react-intl)
const htLocaleData = {
  locale: 'ht',
  pluralRuleFunction: (e, t) => (t && e === 1 ? 'one' : 'other'),
};

const combineTranslationMessages = require('./translations/langs.json');

addLocaleData(enLocaleData);
addLocaleData(esLocaleData);
addLocaleData(ptLocaleData);
addLocaleData(htLocaleData);

const DEFAULT_LOCALE = 'en';

// prettier-ignore
const appLocales = [
  'en',
  'es',
  'pt',
  'ht'
];

const formatTranslationMessages = (locale, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE
      ? formatTranslationMessages(DEFAULT_LOCALE, combineTranslationMessages)
      : {};

  const flattenFormattedMessages = (formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE
        ? defaultFormattedMessages[key]
        : messages[key];
    return Object.assign(formattedMessages, {
      [key]: formattedMessage[locale],
    });
  };
  return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

const translationMessages = {
  en: formatTranslationMessages('en', combineTranslationMessages),
  es: formatTranslationMessages('es', combineTranslationMessages),
  pt: formatTranslationMessages('pt', combineTranslationMessages),
  ht: formatTranslationMessages('ht', combineTranslationMessages),
};

exports.appLocales = appLocales;
exports.formatTranslationMessages = formatTranslationMessages;
exports.translationMessages = translationMessages;
exports.DEFAULT_LOCALE = DEFAULT_LOCALE;
