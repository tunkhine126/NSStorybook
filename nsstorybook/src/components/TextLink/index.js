/**
 *
 * TextLink
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';

import { textLinkStyles, btnStyles } from './styles';

function TextLink({ content, className, disable, bold = true, ...rest }) {
  const classes = textLinkStyles();
  return (
    <Button
      classes={btnStyles(bold)()}
      disabled={disable}
      color="primary"
      className={className || classes.btn}
      {...rest}
    >
      {content}
    </Button>
  );
}

TextLink.propTypes = {
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    .isRequired,
  bold: PropTypes.bool,
  disable: PropTypes.bool,
  className: PropTypes.any,
};

export default TextLink;
