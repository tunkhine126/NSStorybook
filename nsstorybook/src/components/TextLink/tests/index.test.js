/**
 *
 * Tests for TextLink
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import { DEFAULT_LOCALE } from '../../../../i18n';
import TextLink from '../index';

describe('<TextLink />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <TextLink content="Test" />
      </IntlProvider>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <TextLink content="Test" />
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
