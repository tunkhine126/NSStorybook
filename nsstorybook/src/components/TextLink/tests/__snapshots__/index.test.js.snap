// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLink /> Should render and match the snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-46 MuiButton-textPrimary makeStyles-textPrimary-48"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label makeStyles-label-47"
  >
    Test
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
